web/src/components/merchant/sign-up/SignUpCard.tsx

```diff
@@ -118,24 +118,50 @@ const SignUpCard: React.FC = () => {
     // If both passwords are empty we show PASSWORD_EMPTY.
     event.target.value === password
       ? setConfirmPasswordError('')
-      : setConfirmPasswordError(PASSWORDS_DO_NOT_MATCH);
+      : setConfirmPasswordError(Errors.PASSWORDS_DO_NOT_MATCH);
     setConfirmPassword(event.target.value);
   };
 
   const handleVpaChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    event.target.value ? setVpaError('') : setVpaError(VPA_EMPTY);
+    event.target.value ? setVpaError('') : setVpaError(Errors.VPA_EMPTY);
     setVpa(event.target.value);
   };
 
   const handleSubmit = () => {
     // These checks are needed because empty error in the handleChange is shown
     // only if user already fills in the field and then deletes it.
-    if (!name) setNameError(NAME_EMPTY);
-    if (!email) setEmailError(EMAIL_EMPTY);
-    if (!password) setPasswordError(PASSWORD_EMPTY);
+    if (!name) setNameError(Errors.NAME_EMPTY);
+    if (!email) setEmailError(Errors.EMAIL_EMPTY);
+    if (!password) setPasswordError(Errors.PASSWORD_EMPTY);
     if (password !== confirmPassword)
-      setConfirmPasswordError(PASSWORDS_DO_NOT_MATCH);
-    if (!vpa) setVpaError(VPA_EMPTY);
+      setConfirmPasswordError(Errors.PASSWORDS_DO_NOT_MATCH);
+    if (!vpa) setVpaError(Errors.VPA_EMPTY);
+
+    if (!name || !email || !password || password !== confirmPassword || !vpa) {
+      return;
+    }
+
+    handleFirebaseSignUp();
+  };
+
+  /**
+   * This adds a new user account to Firebase so that user can be signed in with
+   * Firebase Authentication in the future.
+   */
+  const handleFirebaseSignUp = async () => {
+    try {
+      await firebaseAuth.createUserWithEmailAndPassword(email, password);
+    } catch (error) {
+      if (error.code === 'auth/invalid-email') {
```

Comment:
Can change to switch statement?

"Generally speaking, if-else is best used when there are two discrete values or a few different ranges of values for which to test. When there are more than two discrete values for which to test, the switch statement is the most optimal choice" - [source](https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html#if-else_versus_switch)

Reply:
Okay! Yep, switch-case is better in this case.